### PR TITLE
show course progress in user profile

### PR DESCRIPTION
**Pull request by VuongLong**

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -67,3 +67,20 @@ en:
...
+    in_progress: "In-progress course"
+    finished: "Finish courses"
+  user_task:
+    name: "Name:"
+    description: "Description:"
+    status: "Status:"
+  status:
+    start: "start"
+    progress: "progress"
+    finished: "finished"
```
Comment (hienbx94) [inline config/locales/en.yml]: line

File: app/views/users/_inprogress_course.html.erb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            <% else %>
+              <span class="label label-info">
+                <%= t "status.progress" %>
+              </span>
+            <% end %>
+          </dd>
+        </dl>
+      </div>
+  
+
```
Comment (hienbx94) [inline app/views/users/_inprogress_course.html.erb]: remove line

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -2,7 +2,17 @@ class UsersController < ApplicationController
   load_and_authorize_resource
 
   def show
-    @activities = PublicActivity::Activity.user_activities(@user.id).recent.decorate
+    @activities = PublicActivity::Activity.user_activities(@user.id).recent.limit(20).decorate
+    @user_course = @user.user_courses.actived.last
```
Comment (hienbx94) [inline app/controllers/users_controller.rb]: check

File: app/views/courses/show.html.erb [hunk 4]
```diff
@@ -66,7 +66,7 @@
       <% end %>
     </div>
   <% end %>
-
+
```
Comment (minhld1709) [inline app/views/courses/show.html.erb]: iden

File: app/views/users/_course.html.erb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  <dd>
+    <small class="text-muted">
+      <%= course.created_at.strftime Settings.date_format %>
+    </small>
+  </dd>
+  
+  <dt><strong><%= t "courses.end_date" %></strong></dt>
+  <dd>
+    <small class="text-muted">
+      <%#= course.created_at.strftime Settings.date_format %>
```
Comment (minhld1709) [inline app/views/users/_course.html.erb]: remove if now dont use

File: app/views/users/_inprogress_course.html.erb (hunk 2 above)
Comment (trantube94) [inline app/views/users/_inprogress_course.html.erb]: remove line

File: app/views/users/_inprogress_course.html.erb [hunk 6]
```diff
@@ -0,0 +1,40 @@
+<%= render "course", course: course %>
+
+<% @user_subjects.each do |user_subject| %>
+    <div class="course_subject">
```
Comment (trantube94) [inline app/views/users/_inprogress_course.html.erb]: indent

File: config/locales/en.yml (hunk 1 above)
Comment (trantube94) [inline config/locales/en.yml]: add line

File: app/views/users/show.html.erb [hunk 7]
```diff
@@ -1,26 +1,62 @@
...
+          <% @finished_courses.each do |course| %>
+            <li class="list-group-item finished_course">
+              <%= render "course", course: course %>
+            </li>
+          <% end %>
+        </div>
+      </div>
+    </div>
   </div>
 </div>
```
Comment (trantube94) [inline app/views/users/show.html.erb]: remove line

File: app/views/users/_course.html.erb [hunk 8]
```diff
@@ -0,0 +1,28 @@
+<dl class="dl-horizontal">
+  <dt><strong><%= t "courses.name" %></strong></dt>
+  <dd><%= course.name %></dd>
+  
+  <dt><strong><%= t "courses.description" %></strong></dt>
+  <dd><%= course.description %></dd>
+  
+  <dt><strong><%= t "courses.status" %></strong></dt>
+  <% if course.progress?  %>
```
Comment (hellovietnam93) [inline app/views/users/_course.html.erb]: 1 dòng thôi e ơi

File: app/models/user_course.rb [hunk 9]
```diff
@@ -16,7 +16,6 @@ class UserCourse < ActiveRecord::Base
   has_many :user_subjects, dependent: :destroy
 
   scope :actived, ->{where active: true}
-
```
Comment (trantube94) [inline app/models/user_course.rb]: dont remove this line

File: app/controllers/users_controller.rb [hunk 10]
```diff
@@ -2,7 +2,18 @@ class UsersController < ApplicationController
   load_and_authorize_resource
 
   def show
-    @activities = PublicActivity::Activity.user_activities(@user.id).recent.decorate
+    @activities = PublicActivity::Activity.user_activities(@user.id).recent.limit(20).decorate
+    if @user_course = @user.user_courses.actived.lastF
```
Comment (trantube94) [inline app/controllers/users_controller.rb]: ???

**Comment by hienbx94** [comment]
OK

File: app/views/users/show.html.erb [hunk 11]
```diff
@@ -1,26 +1,65 @@
...
+      <div class="tab-content">
+        <div id="inprogress_course" class="tab-pane active">
+          <li class="list-group-item finished_course">
+            <% if @inprogress_course.present? %>
+              <%= render "inprogress_course", course: @inprogress_course %>
+            <% end %>
+          </li>
+        </div>
+        <div id="finished_course" class="tab-pane fade">
+          <% if  %>
```
Comment (trantube94) [inline app/views/users/show.html.erb]: ???

**Comment by hellovietnam93** [comment]
OK

**Comment by ttuan** [comment]
OK

**Comment by trantube94** [comment]
OK

**Pull request merged**